### PR TITLE
cliccl: add `load show backups` to display backup collection

### PR DESCRIPTION
**Pull request by Elliebababa**
Previously, users can list backups created by `BACKUP INTO`
with `SHOW BACKUP IN`in a sql session. But this listing task
can be also done offline without a running cluster.

This PR updates `load show` with `backups` subcommand, 
which allows users to list backups in a backup collection 
created by `BACKUP INTO`. 
With the same purpose as other `load show` subcommands, 
this update allows users to list backups without running 
`SHOW BACKUP IN` in a sql session.

see #61131 #61829 to checkout other `load show` subcommand.

Release note (cli change): Add `load show backups` to
display backup collection. Previously, users can list backups
created by `BACKUP INTO` via `SHOW BACKUP IN`in a sql
session. But this listing task can be also done offline without a
running cluster. Now, users are able to list backups in a collection
with `cockroach load show backups <collection_url>.

### CODE REVIEW

File: pkg/ccl/cliccl/load.go [hunk 1]
```diff
@@ -122,6 +132,35 @@ func runLoadShowSummary(cmd *cobra.Command, args []string) error {
...
+		return err
+	}
+
+	backupPaths, err := store.ListFiles(ctx, "/*/*/*/"+backupccl.BackupManifestName)
+	if err != nil {
+		return err
+	}
+
+	if len(backupPaths) == 0 {
+		fmt.Println("no backups found.")
```
Comment (pbardea) [inline pkg/ccl/cliccl/load.go:154]: Can we add a test that exercises this?

File: pkg/ccl/cliccl/load.go [hunk 2]
```diff
@@ -122,6 +132,35 @@ func runLoadShowSummary(cmd *cobra.Command, args []string) error {
...
+		path = cloudimpl.MakeLocalStorageURI(path)
+	}
+	ctx := context.Background()
+	store, err := cloudimpl.ExternalStorageFromURI(ctx, path, base.ExternalIODirConfig{},
+		cluster.NoSettings, newBlobFactory, security.RootUserName(), nil /*Internal Executor*/, nil /*kvDB*/)
+	if err != nil {
+		return err
+	}
+
+	backupPaths, err := store.ListFiles(ctx, "/*/*/*/"+backupccl.BackupManifestName)
```
Comment (pbardea) [inline pkg/ccl/cliccl/load.go:148]: nit: Could we move this into a very small method inside `backupccl` (maybe in `backup_destinations`?) that keeps the internal format of the backup inside `backupccl`. I'm thinking something like `backupccl.ListFullBackupsInCollection(ctx, store)`. Although it's a small wrapper, I would like to keep as much of the specifics of how backups are stored inside `backupccl` and limit visibility into the backup format with a more structured API.

**Review by pbardea — COMMENTED** [review]
I think a couple of my comments on the incremental PR also apply here, but mostly nits.

**Review by pbardea — APPROVED** [review]
Second commit LGTM

**Comment by Elliebababa** [comment]
TFTR!

bors r=pbardea

**Pull request merged**